plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2912,6 +2949,80 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 Optional.of(physicalColumnNameMapping));
     }
 
+    private void generateMissingFileStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = transactionLogAccess.getActiveFiles(
+                        getSnapshot(tableHandle.getSchemaTableName(), tableHandle.getLocation(), session), session)
+                .stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getNumRecords().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                .filter(addFileEntry -> !URI.create(addFileEntry.getPath()).isAbsolute()) //Support only relative path in AddFileEntry
+                // Statistics returns whole path to file, so we need to create corresponding map key for AddFileEntry.
+                .collect(toImmutableMap(addFileEntry -> Location.of(tableHandle.getLocation()).appendPath(addFileEntry.getPath()).toString(), identity()));
```

Comment:
The logic to build the file path is different from `DeltaLakeSplitManager#buildSplitPath`. This will generate wrong keys when the path contains special characters.